### PR TITLE
added missing mesheryctl guides

### LINKED ISSUE #4935: [Docs] Add all `mesheryctl` guides to bulleted collection
#### Current State
This page category for mesheryctl guides (https://docs.meshery.io/guides/mesheryctl) is missing a number of mesheryctl guides.

Screen Shot 2022-01-12 at 8 45 32 PM

#### Desired State

All of the mesheryctl guides should be listed on the page.

---

#### [Contributor Guides](https://docs.meshery.io/project/contributing) and Resources
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Vidyaa123**
#Signed-off-by: Vidyaa <mailvidyaa@gmail.com>

**Description**

This PR fixes  #4935

**Notes for Reviewers**
added missing mesheryctl guides and fixed wrong permalinks for two guides

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 220ef25eed81491cec5b75afc4cbd588df5ea563 at: https://61e03c89ce4455945e900fa8--meshery-docs-preview.netlify.app

**Review by verma-kunal — DISMISSED** [review]
lgtm 👍  Great work @Vidyaa123

**Review by Chadha93 — APPROVED** [review]
LGTM, good work @Vidyaa123 :1st_place_medal:

**Comment by Vidyaa123** [comment]
> LGTM, good work @Vidyaa123 🥇


> lgtm 👍 Great work @Vidyaa123



> lgtm 👍 Great work @Vidyaa123

Thank You! @verma-kunal

**Comment by Vidyaa123** [comment]
> lgtm 👍 Great work @Vidyaa123

Thank you! @verma-kunal

**Comment by Vidyaa123** [comment]
> LGTM, good work @Vidyaa123 🥇

Thanks for encouraging! @Chadha93

**Comment by leecalcote** [comment]
@Vidyaa123 I wonder if you might like to try the Jekyll challenge that was presented on a couple of our recent calls. Here's a description from the Meshery call this week - https://youtu.be/eU72fb0bHdA?t=1578

File: docs/pages/guides/mesheryctl/index.md [hunk 1]
```diff
@@ -12,8 +12,11 @@ list: exclude
 Guides to using Meshery's various features and components.
 
 {% capture tag %}
-
-<li><a href="{{ site.baseurl }}/guides/upgrade#upgrading-meshery-cli">Upgrading mesheryctl</a></li>
+<li><a href="{{ site.baseurl }}/guides/mesheryctl/working-with-mesheryctl">Using mesheryctl</a></li>
```
Comment (leecalcote) [inline docs/pages/guides/mesheryctl/index.md:15]: @Vidyaa123, thanks for taking on this issue. Quality work.

Let's make this dynamic and less fragile. Will you please consider a query for `type: Guides` of `category:mesheryctl`?

**Review by leecalcote — CHANGES_REQUESTED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit 671ef8eb84d51702e24f65dd314e49297788965f at: https://61e12c78a3cde2d752bc14d1--meshery-docs-preview.netlify.app

File: docs/pages/guides/mesheryctl/index.md (hunk 1 above)
Comment (Vidyaa123) [reply docs/pages/guides/mesheryctl/index.md]: @leecalcote, I have added dynamic links to the mesheryctl guides except upgrading which links to a header. Also, the order of the mesheryctl guides is different to the side navigation. Should I add an order to the guides and sort them based on that.

**Comment by Vidyaa123** [comment]
> @Vidyaa123 I wonder if you might like to try the Jekyll challenge that was presented on a couple of our recent calls. Here's a description from the Meshery call this week - https://youtu.be/eU72fb0bHdA?t=1578

@leecalcote, I will check the call. 
I am very new to Jekyll but up for a challenge though.

File: docs/pages/guides/mesheryctl/index.md (hunk 1 above)
Comment (leecalcote) [reply docs/pages/guides/mesheryctl/index.md]: @Vidyaa123 sorting alphabetically will do the trick.

**Comment by leecalcote** [comment]
> @Vidyaa123 I wonder if you might like to try the Jekyll challenge that was presented on a couple of our recent calls. Here's a description from the Meshery call this week - https://youtu.be/eU72fb0bHdA?t=1578

@Vidyaa123, the gauntlet has been laid down... what do you think?

**Comment by leecalcote** [comment]
@shubham-tyagi, where you at? ;)

**Comment by shubham-tyagi** [comment]
@leecalcote I have been assigned #3765, after solving that I would for sure look into what you asked for in the meeting. I have been busy somewhere else for a while but I have started to solve #3765.

**Comment by leecalcote** [comment]
> @leecalcote I have been assigned #3765, after solving that I would for sure look into what you asked for in the meeting. I have been busy somewhere else for a while but I have started to solve #3765.

Thank you for this update. Sounds real good.

File: docs/pages/guides/mesheryctl/working-with-mesheryctl.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 layout: default
 title: Using mesheryctl
 description: How to use mesheryctl
-permalink: guides/mesheryctl/using-mesheryctl
+permalink: guides/mesheryctl/working-with-mesheryctl
```
Comment (leecalcote) [inline docs/pages/guides/mesheryctl/working-with-mesheryctl.md:5]: @Vidyaa123 will you confirm that this doesn't break another internal link to the previous `using-mesheryctl` permalink?

**Comment by l5io** [comment]
🚀 Preview for commit 4ae374f40606bc183d4230f43e424c42080290d8 at: https://61e40bef9f4ef3cc5c4f84e8--meshery-docs-preview.netlify.app

File: docs/pages/guides/mesheryctl/working-with-mesheryctl.md (hunk 2 above)
Comment (Vidyaa123) [reply docs/pages/guides/mesheryctl/working-with-mesheryctl.md:5]: @leecalcote, I have missed this. It does break the 'using-mesheryctl'  and 'running-system-checks using mesheryctl' internal link as well. The urls are hardcoded in toc.yml file.  I have fixed that now. and it works for me. Please review. Thanks.

**Comment by warunicorn19** [comment]
Hey @Vidyaa123 would you be available to discuss this PR on tomorrow's websites' call?

**Comment by leecalcote** [comment]
@Vidyaa123 you were missed. ;)

File: docs/pages/guides/mesheryctl/working-with-mesheryctl.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 layout: default
 title: Using mesheryctl
```
Comment (leecalcote) [inline docs/pages/guides/mesheryctl/working-with-mesheryctl.md:3]:
```suggestion
title: Using Meshery CLI
```

**Comment by leecalcote** [comment]
@Vidyaa123 still working on alphabetical sorting of the list of guides?

**Comment by l5io** [comment]
🚀 Preview for commit 31d27b4d551f051d745c6b63ebb82a3c81ec9fc9 at: https://61e5862798345c945983bbd7--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 36a151f28f73b30b58033e4977e654af5e37e8bb at: https://61e6582c987abfb7137506dd--meshery-docs-preview.netlify.app

**Comment by Vidyaa123** [comment]
> @Vidyaa123 still working on alphabetical sorting of the list of guides?

@leecalcote, sorted them alphabetically now.

**Comment by Vidyaa123** [comment]
> Hey @Vidyaa123 would you be available to discuss this PR on tomorrow's websites' call?

@warunicorn19, Apologies, couldn't join the call. I had a clash. Is there anything I had to look at?

**Comment by Vidyaa123** [comment]
> > @Vidyaa123 I wonder if you might like to try the Jekyll challenge that was presented on a couple of our recent calls. Here's a description from the Meshery call this week - https://youtu.be/eU72fb0bHdA?t=1578
> 
> @Vidyaa123, the gauntlet has been laid down... what do you think?

I think I'll try this one.

**Review by Abhijay007 — APPROVED** [review]
LGTM 👍 

only DCO left

**Review by leecalcote — APPROVED** [review]
Looks great, @Vidyaa123! 🎈

File: docs/pages/guides/mesheryctl/index.md [hunk 4]
```diff
@@ -11,10 +11,17 @@ list: exclude
 
 Guides to using Meshery's various features and components.
 
-{% capture tag %}
+{% assign sorted_guides = site.pages | sort: "name" %}
+<ul>
+  {% for item in sorted_guides %}
```
Comment (leecalcote) [inline docs/pages/guides/mesheryctl/index.md:16]: @Vidyaa123 this doesn't seem to be working in the docs. We're missing this list, it seems.

Comment (leecalcote) [reply docs/pages/guides/mesheryctl/index.md:16]: Never mind! I was looking at an old branch. We're all set here. 👍

**Pull request merged**